tools/convert_tabular_data_to_fhir-death-record.rb

```diff
@@ -0,0 +1,324 @@
+    # Sometimes the date of month is in the form "Ap", "Ma". Need to take the full `Injury date` field to get the proper month data.
+    ije_field_string = convert_month_str(data_row['Injury date'])
+  elsif ije_key == 'DSTATE' && ije_field_string == ""
+    # Towards the end of the dataset, they stopped including DSTATE for some reason. I'm adding back as "WA".
+    ije_field_string = "WA"
+  end
+  return ije_field_string
+end
+
+# Converts a 2-char month string to a number. Obviously, there are overalapping months but that's how it is in the data.
+def convert_month_str(string)
+  if string.include?("Jan")
+    return "1"
+  elsif string.include?("Feb")
+    return "2"
+  elsif string.include?("Mar")
+    return "3"
+  elsif string.include?("Apr")
+    return "4"
+  elsif string.include?("May")
+    return "5"
+  elsif string.include?("Jun")
+    return "6"
+  elsif string.include?("Jul")
+    return "7"
+  elsif string.include?("Aug")
+    return "8"
+  elsif string.include?("Sep")
+    return "9"
+  elsif string.include?("Oct")
+    return "10"
+  elsif string.include?("Nov")
+    return "11"
+  elsif string.include?("Dec")
+    return "12"
+  end
+end
+
+# Maps the ethncitiy value to an IJE-accepted format.
+def map_ethnicity_value(value)
+  if value == 'N'
+    return 'N'
+  elsif value == 'Y'
+    return 'H'
+  end
+  return 'U'
+end
+
+# Returns whether a given string is numeric.
+def is_number?(string)
+  true if Float(string) rescue false
+end
+
+# Formats the given ije data based on the length and whether it is a number or text.
+def format_ije_data(ije_field_string, ije_length)
+  # Truncate and warn of overlong IJE field values.
+  if ije_field_string.length > ije_length
+    # puts "Record `#{data_row['State file number']}` contains an overlong data field `#{ije_field_string}` for `#{ije_key}`. Expected `#{ije_length}`. Truncating."
+    ije_field_string = ije_field_string[0,ije_length]
+  end
+  if ije_field_string == "."
+    ije_field_string = ""
+  end
+  # Add padding spaces to the of the string according to its length.
+  if is_number?(ije_field_string)
+    # Numbers are right justified with 0s.
+    ije_field_string = ije_field_string.rjust(ije_length, '0')
+  else
+    # Strings are left justified.
+    # But it should be automatic, no need to manually do this.
+    ije_field_string = ije_field_string.ljust(ije_length, ' ')
+  end
+end
+
+# Converts an input data sheet into IJE records based on the given mappings of IJE->datafile_headers.
+def convert_ije_records(data_sheet, ije_data_mappings, data_lit)
+
+  # Only so many records at a time can be export via the vrdr CLI.
+  records_before_export = 15
+
+  # Array of IJE record strings.
+  ije_records = Array.new { '' }
+  # Skip column headers row flag.
+  skip_headers = true
+  # Iterate over the input data sheet which contains all the excel-based death records.
+  data_sheet.simple_rows.each do |data_row|
+
+    if ije_records.length() > records_before_export
```

Comment:
Will convert after the conversion occurs.
There's a limit of 15 probably because I'm passing the entire raw IJE strings rather than record names. See my above comment asking if this makes the most sense as the way to do it.

Reply:
Changed to use ije files rather than raw strings.